pkg/build/api/types.go

```diff
@@ -425,6 +429,18 @@ type SourceBuildStrategy struct {
 	ForcePull bool
 }
 
+// JenkinsPipelineStrategy holds parameters specific to a Jenkins Pipeline build.
+// This strategy is experimental.
+type JenkinsPipelineBuildStrategy struct {
+	// JenkinsfilePath is the optional path of the Jenkinsfile that will be used to configure the pipeline
+	// relative to the root of the source repository. If both JenkinsfilePath & Jenkinsfile are
+	// both not specified, this defaults to Jenkinsfile in the root of the specified git repo.
+	JenkinsfilePath string
```

Comment:
Should we just reuse contextDir instead of adding this field?  @smarterclayton ?

we already use contextDir to help us locate a Dockerfile in a git repo, for example.